gkr_iop/src/lib.rs

```diff
@@ -62,39 +67,46 @@ where
         &self,
         circuit: &GKRCircuit<E>,
         phase1_witness_group: &RowMajorMatrix<E::BaseField>,
+        fixed: &[Vec<E::BaseField>],
         challenges: &[E],
     ) -> (GKRCircuitWitness<'a, E>, GKRCircuitOutput<E>) {
         // layer order from output to input
+        let size = phase1_witness_group.num_instances();
         let mut layer_wits = Vec::<LayerWitness<E>>::with_capacity(circuit.layers.len() + 1);
         let phase1_witness_group = phase1_witness_group
             .to_mles()
             .into_iter()
             .map(Arc::new)
             .collect_vec();
 
-        layer_wits.push(LayerWitness::new(phase1_witness_group.clone()));
         let mut witness_mle_flatten = vec![None; circuit.n_evaluations];
 
         // initialize a vector to store the final outputs of the GKR circuit.
         // these outputs correspond to the evaluations at the last layer of the circuit.
         // we preallocate the vector with exact capacity for efficiency, avoiding reallocations.
         // the number of expected outputs is given by `circuit.n_evaluations`.
-        let mut gkr_out_well_order = Vec::with_capacity(circuit.n_evaluations);
+        let mut gkr_out_well_order = Vec::with_capacity(circuit.n_out_evals);
 
         // set input to witness_mle_flatten via first layer in_eval_expr
-        let mut phase1_wit_map = HashMap::new();
-        circuit
-            .openings
-            .iter()
-            .for_each(|(phase1_wit_id, eval)| match eval {
-                EvalExpression::Zero => {}
-                EvalExpression::Single(eval_id) | EvalExpression::Linear(eval_id, _, _) => {
-                    if !phase1_wit_map.contains_key(eval_id) {
-                        phase1_wit_map.insert(*eval_id, (*phase1_wit_id, eval));
-                    }
+        if let Some(first_layer) = circuit.layers.last() {
+            let it = first_layer.in_eval_expr.iter();
```

Comment:
we can omit intermediate var `it`

Reply:
Fixed in https://github.com/scroll-tech/ceno/pull/970/commits/5682bd995729ad270e9a4995fbf3f6880ad9ec0b.

Reply:
Fixed in https://github.com/scroll-tech/ceno/pull/970/commits/5682bd995729ad270e9a4995fbf3f6880ad9ec0b.